client/src/main/scala/skuber/api/WatchSource.scala

```diff
@@ -0,0 +1,121 @@
+                                 name: Option[String],
+                                 sinceResourceVersion: Option[String],
+                                 maxWatchRequestTimeout: Duration,
+                                 bufSize: Int)(implicit sys: ActorSystem,
+                                               fm: Materializer,
+                                               format: Format[O],
+                                               rd: ResourceDefinition[O],
+                                               lc: LoggingContext): Source[WatchEvent[O], NotUsed] = {
+    Source.fromGraph(GraphDSL.create() { implicit b =>
+      import GraphDSL.Implicits._
+
+      implicit val dispatcher: ExecutionContextExecutor = sys.dispatcher
+
+      val initSource = Source.single(
+        (Request[O](sinceResourceVersion), createRequest(rc, name, sinceResourceVersion, rd, maxWatchRequestTimeout))
+      )
+
+      val httpFlow: Flow[(Request[O], HttpRequest), StreamElement[O], _] =
+        Flow[(Request[O], HttpRequest)].mapAsync(1) { request =>
+          rc.invoke(request._2, watch = true).map(response => (request._1, response))
+        }.mapMaterializedValue(_ => NotUsed).flatMapConcat {
+          case (r, HttpResponse(StatusCodes.OK, _, entity, _)) =>
+            val start: Source[StreamElement[O], _] = Source.single(r)
+            val results: Source[StreamElement[O], _] = bytesSourceToWatchEventSource(entity.dataBytes, bufSize)
+            val end: Source[StreamElement[O], _] = Source.single(End[O]())
+            start ++ results ++ end
+          case (r, HttpResponse(sc, _, entity, _)) =>
+            throw new K8SException(Status(message = Some("Error watching resource"), code = Some(sc.intValue())))
+        }
+
+      val outboundFlow: Flow[StreamElement[O], WatchEvent[O], NotUsed] =
+        Flow[StreamElement[O]]
+          .filter(_.isInstanceOf[Result[O]])
+          .map{
+            case Result(_, event) => event
+            case _ => throw new K8SException(Status(message = Some("Error processing watch events.")))
+          }
+
+      val feedbackFlow: Flow[StreamElement[O], (Request[O], HttpRequest), NotUsed] =
+        Flow[StreamElement[O]].scan(StreamContext(None, Waiting)){(cxt, next) =>
+          next match {
+            case Request(rv) => StreamContext(rv, Processing)
+            case Result(rv, _) => StreamContext(Some(rv), Processing)
+            case End() => cxt.copy(state = Finished)
+          }
+        }.filter(_.state == Finished).map { acc =>
+          (Request[O](acc.currentResourceVersion), createRequest[O](rc, name, acc.currentResourceVersion, rd, maxWatchRequestTimeout))
+        }
+
+      val init = b.add(initSource)
+      val http = b.add(httpFlow)
+      val merge = b.add(Merge[(Request[O], HttpRequest)](2))
+      val broadcast = b.add(Broadcast[StreamElement[O]](2))
+      val outbound = b.add(outboundFlow)
+      val feedback = b.add(feedbackFlow)
+
+      // format: OFF
```

Comment:
Currently don't use Scalafmt with this project, so I'd say the format instructions should be removed as it would look odd otherwise.